video2dataset/subsamplers/caption_subsampler.py

```diff
@@ -0,0 +1,183 @@
+"""
+video captioner
+"""
+import os
+import glob
+import json
+import argparse
+import string
+from pathlib import Path
+from functools import partial
+
+import torch
+import torch.nn as nn
+from einops import rearrange, repeat
+
+from transformers import (
+    AutoModelForCausalLM,
+    AutoModelForSeq2SeqLM,
+    BatchEncoding,
+    Blip2Config,
+    Blip2ForConditionalGeneration,
+    Blip2Processor,
+    Blip2QFormerModel,
+    Blip2VisionModel,
+    AutoTokenizer,
+    BitsAndBytesConfig
```

Comment:
I don't understand why we need all of these classes + the 3 classes you define. I thought there would be like One class that takes in a video and outputs a string, is there not?

Reply:
cleaned up the imports to remove unused ones!